src/test/test-fs-util.c

```diff
@@ -846,6 +846,26 @@ static void test_chmod_and_chown_unsafe(void) {
         assert_se(S_ISLNK(st.st_mode));
 }
 
+static void test_path_is_encrypted_one(const char *p, int expect) {
+        int r;
+
+        r = path_is_encrypted(p);
+        assert_se(r >= 0);
+
+        printf("%s encrypted: %s\n", p, yes_no(r));
+
+        assert_se(expect < 0 || ((r > 0) == (expect > 0)));
+}
+
+static void test_path_is_encrypted(void) {
+        test_path_is_encrypted_one("/home", -1);
+        test_path_is_encrypted_one("/var", -1);
+        test_path_is_encrypted_one("/", -1);
+        test_path_is_encrypted_one("/proc", false);
+        test_path_is_encrypted_one("/sys", false);
+        test_path_is_encrypted_one("/dev", false);
```

Comment:
Hmm, this doesn't seem to work at all:
```console
$ build/test-fs-util
...
/home encrypted: no
/var encrypted: no
/ encrypted: no
/proc encrypted: no
/sys encrypted: no
/dev encrypted: no
```
and the same with `sudo`. I have full-disk encryption with LVM. I guess this is more complicated to discover, but it is a very popular setup and inability to detect this limits the usefulness of this check significantly.

Reply:
so this would be relatively easy to get working. I have no LVM and didn't want to set it up, that's all. I couldn't test it without...

When determining a block device is not encrypted, path_is_encrypted() should iterate through /sys/dev/block/%u:%u/slaves/, which lists all backing devices of the block device if it exists. It should then just run the original test again on all listed block devices, and return true if all of them are encrypted. As extra safety precaution it might make sense to not descend endlessly though, but put a limit on it.

But maybe let's do that in a separate PR.

Reply:
OK.

Reply:
I added (lightly tested) support for such complex storage in #15744, ptal!